services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -2606,11 +2608,25 @@ private void addSupersetSchemaForStore(String clusterName, String storeName, boo
 
   private void addWriteComputeSchemaForStore(String clusterName, String storeName, boolean dryRun) {
     Collection<SchemaEntry> valueSchemaEntries = getValueSchemas(clusterName, storeName);
+
     List<SchemaEntry> writeComputeSchemaEntries = new ArrayList<>(valueSchemaEntries.size());
+    int maxId = valueSchemaEntries.stream().map(SchemaEntry::getId).max(Comparator.naturalOrder()).get();
+
     for (SchemaEntry valueSchemaEntry: valueSchemaEntries) {
-      Schema writeComputeSchema =
-          writeComputeSchemaConverter.convertFromValueRecordSchema(valueSchemaEntry.getSchema());
-      writeComputeSchemaEntries.add(new SchemaEntry(valueSchemaEntry.getId(), writeComputeSchema));
+      Schema writeComputeSchema = null;
+      try {
+        writeComputeSchema = writeComputeSchemaConverter.convertFromValueRecordSchema(valueSchemaEntry.getSchema());
+      } catch (VeniceException e) {
+        // Allow write compute schema error in all schema except the latest value schema
+        if (valueSchemaEntry.getId() == maxId) {
+          throw new VeniceException(
+              "Cannot generate update schema for value schema, schema missing defaults." + valueSchemaEntry,
+              e);
+        }
+      }
+      if (writeComputeSchema != null) {
```

Comment:
Why not put this piece of logic inside `try` scope?

Reply:
Sure, ideally it should not throw any exception.